crates/nu-command/src/filters/find.rs

```diff
@@ -320,6 +320,29 @@ fn find_with_rest_and_highlight(
     };
     let ls_colors = get_ls_colors(ls_colors_env_str);
 
+    let input = if let PipelineData::Value(Value::String { ref val, span }, _) = input {
+        if val.contains('\n') {
```

Comment:
To be perfectly honest I am not too fond of having an "effectively mutating" `if let` for the special case before the main match. On the plus side it fights the rightward drift (i.e. more and more nesting levels) but on the downside It splits the logic apart a bit. 
Might be worth moving the handling of lines in a small well named function that is invoked when encountering a `PipelineData::Value` in the main match to perform this mapping.

But not the end of the world if you don't have time for a refactoring.